themes/tdr/login/login-config-totp.ftl

```diff
@@ -0,0 +1,78 @@
+<#import "template.ftl" as layout>
+<@layout.registrationLayout displayInfo=true displayRequiredFields=true; section>
+
+    <#if section = "header">
+        ${msg("loginTotpTitle")}
+
+    <#elseif section = "form">
+
+        <ul id="kc-totp-settings" class="govuk-list--number">
+            <li>
+                <label class="govuk-label" for="supported-apps">
+                    ${msg("loginTotpStep1")}
+                </label>
+
+                <ul id="supported-apps" class="govuk-list govuk-list--bullet">
```

Comment:
Did you try setting the list of supported apps, via the config json? Did a search across the code, and this appears to be the only place this list was set:

`...
"otpPolicyPeriod": 30,
  "otpSupportedApplications": [   
    "Google Authenticator",
    "FreeOTP"
  ],
  "webAuthnPolicyRpEntityName": "keycloak",
...`

Reply:
I tried setting that but it doesn't actually do anything. The supported apps are hard coded in the keycloak code and I think it will be difficult/impossible to change. I've left them hard coded for now.